desugreet/bot.py

```diff
@@ -4,44 +4,109 @@
 
 intents = discord.Intents.default()
 intents.members = True
+intents.message_content = True
 
 client = discord.Client(intents=intents)
 token = None
-role = None
+member_role_str = None
 welcome_msg = None
 log_channel_id = None
+entrance_role_str = None
+entrance_channel_id = None
+entrance_msg = None
 start_time = datetime.datetime.now()
 
+entrance_track_buffer = {}
 
 @client.event
 async def on_member_join(member):
     server = member.guild
-    member_role = __getRole(server.roles, role)
-    if member_role is not None:
-        await member.add_roles(member_role)
+    entrance_role = discord.utils.get(server.roles, name=entrance_role_str)
+    if entrance_role is not None:
+        await member.add_roles(entrance_role)
     else:
         print("check your config file, the role does not exist!")
-    await member.send(welcome_msg.format(member, server))
-    await client.get_channel(log_channel_id).send("{0.mention} has joined".format(member))
 
+    entrance_track_buffer[member] = []  # create list buffer with a member obj key
+    entrance_track_buffer[member].append(await client.get_channel(entrance_channel_id).send(entrance_msg.format(member)))
```

Comment:
do we need a try catch for the await there? what happens on failure there? Can probably ignore tho

Reply:
I don't really feel the need for it since the failure occurs on configuration error